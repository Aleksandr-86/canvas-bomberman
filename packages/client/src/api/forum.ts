import axios, { AxiosError} from "axios"

export const ForumApi = {
  async getPosts() {
    try {
      const response = await axios.get(
        'https://alt-f4-bomberman-21.ya-praktikum.tech/posts'
      )
      return response.data
    } catch (e) {
      return e as AxiosError
    }
  },

  async sendPosts(formData: FormData) {
    try {
      const response = await axios.post(
        'https://alt-f4-bomberman-21.ya-praktikum.tech/posts',
        {
          data: formData,
        }
      )
      return response.data
    } catch (e) {
      return e as AxiosError
    }
  },
  async sendLike(id: number) {
    try {
      await axios.post('https://alt-f4-bomberman-21.ya-praktikum.tech/likes', {
        data: id,
      })
      return id
    } catch (e) {
      return e as AxiosError
    }
  },

  async sendDislike(id: number) {
    try {
      await axios.post(
        'https://alt-f4-bomberman-21.ya-praktikum.tech/dislikes',
        {
          data: id,
        }
      )
      return id
    } catch (e) {
      return e as AxiosError
    }
  },
  async getMessages(id: string | undefined) {
    try {
      const response = await axios.get(
        `https://alt-f4-bomberman-21.ya-praktikum.tech/posts/${id}`
      )
      return response.data
    } catch (e) {
      return e as AxiosError
    }
  },

  async sendMessage(formData: FormData) {
    try {
      const response = await axios.post(
        `https://alt-f4-bomberman-21.ya-praktikum.tech/posts/`,
        {
          data: formData,
        }
      )
      return response.data
    } catch (e) {
      return e as AxiosError
    }
  }

}
