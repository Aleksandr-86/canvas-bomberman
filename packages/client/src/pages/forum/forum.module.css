.forum {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.forumWrapper {
  display: flex;
  flex-direction: column;
}

.forumTitle {
  font-weight: 400;
  font-size: 44px;
  font-family: 'Rubik Bubbles', fantasy;
  line-height: 49px;
  text-align: center;
}

.forumLink {
  max-width: 250px;
  margin-bottom: 30px;
  padding: 10px 50px;
  border: 5px solid black;
  border-radius: 15px;
  background: hsl(29deg 6% 89%);
  color: black;
  font-family: 'Rubik Bubbles', fantasy;
  text-transform: uppercase;
  cursor: pointer;
  transition: all 0.2s linear;
}
