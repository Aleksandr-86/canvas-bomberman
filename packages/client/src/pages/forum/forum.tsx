import { Link } from 'react-router-dom'
import classes from './forum.module.css'
import { Table } from '../../components/table/table'
import { useAppDispatch, useAppSelector } from '../../store/hooks'
import { getPosts as getAll } from '../../store/forumSlice'
import { getPosts } from '../../store/selectors'
import { useEffect } from 'react'

export const Forum = () => {
  const dispatch = useAppDispatch()
  useEffect(() => {
    dispatch(getAll)
  }, [])
  const posts = useAppSelector(getPosts)
  return (
    <div className={classes.forum}>
      <h2 className={classes.forumTitle}>Форум</h2>
      <Table {...posts} />
      <div className={classes.forumWrapper}>
        <Link to="/forum-new-post" className={classes.forumLink}>
          Создать тему
        </Link>
        <Link to="/" className={classes.forumLink}>
          на главную
        </Link>
      </div>
    </div>
  )
}
