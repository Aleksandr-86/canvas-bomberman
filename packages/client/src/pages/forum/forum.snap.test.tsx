import * as renderer from 'react-test-renderer'
import { Forum } from './forum'
import { MemoryRouter } from 'react-router-dom'

describe(`Snapshot test page Forum`, () => {
  test(`Should correct render (without props)`, () => {
    const tree = renderer
      .create(
        <MemoryRouter initialEntries={[`/forum`]}>
          <Forum />
        </MemoryRouter>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
