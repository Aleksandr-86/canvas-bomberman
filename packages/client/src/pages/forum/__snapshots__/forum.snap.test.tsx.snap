// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test page Forum Should correct render (without props) 1`] = `
<div
  className="forum"
>
  <h2
    className="forumTitle"
  >
    Форум
  </h2>
  <table
    className="table"
  >
    <thead>
      <tr>
        <th
          className="tableTitle"
        >
          Автор
        </th>
        <th
          className="tableTitle"
        >
          Тема
        </th>
        <th
          className="tableTitle"
        >
          Дата
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          className="tableRow"
        >
          Kan88
        </td>
        <td
          className="tableRow"
        >
          Как пройти первый уровень
        </td>
        <td
          className="tableRow"
        >
          8.01.22
        </td>
      </tr>
      <tr>
        <td
          className="tableRow"
        >
          sasha85
        </td>
        <td
          className="tableRow"
        >
          Как создать новую тему
        </td>
        <td
          className="tableRow"
        >
          8.01.22
        </td>
      </tr>
    </tbody>
  </table>
  <div
    className="forumWrapper"
  >
    <a
      className="forumLink"
      href="/forum-new-post"
      onClick={[Function]}
    >
      Создать тему
    </a>
    <a
      className="forumLink"
      href="/"
      onClick={[Function]}
    >
      на главную
    </a>
  </div>
</div>
`;
